embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/injection/InitModule.kt

```diff
@@ -16,6 +17,11 @@ internal interface InitModule {
      */
     val clock: Clock
 
+    /**
+     * Service to track usage of public APIs and other internal metrics
+     */
+    val embraceTelemetryService: EmbraceTelemetryService
```

Comment:
Consider creating a `TelemetryService` interface that defines the functions that form this service's internal API. This will make testing much easier as we can create fake/no-ops & lets folks know what is the internal API & what isn't

Reply:
Do you always create the interfaces by default? I usually wait until I actually need a fake or other implementation, but I can add it, no problem!

Reply:
I usually try to create one by default - the benefit looks a bit more marginal here as the service doesn't take any params etc

Reply:
Agreed with creating the interface for anything we are calling a Service. We almost always want to create a fake for tests given the way our modules and instantiation works, and doing so also forces us to think about what logic needs to expose externally vs things that should be kept internal, especially when the implementation evolves.